batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/internal/contentassist/BatchJobCompletionProposalComputer.java

```diff
@@ -263,4 +292,51 @@ Node findNodeForOffset(IDOMNode node, int offset) {
 		return node;
 	}
 
-}
+	private void addClassValueProposals(final ContentAssistRequest contentAssistRequest, CompletionProposalInvocationContext context, 
+			Element current, String matchString, final int begin, final int length) throws JavaModelException {
+		if (matchString.length()>0) {
+			int lastDot = matchString.lastIndexOf('.');
+			String packageNameString = null;
+			String classNameString = matchString;
+			int packagePatern = SearchPattern.R_PREFIX_MATCH;
+			if(lastDot>0) {
+				if(matchString.length()>lastDot+1 && Character.isUpperCase(matchString.charAt(lastDot+1))) {
+					classNameString = matchString.substring(lastDot+1);
+					packageNameString = matchString.substring(0, lastDot);
+					packagePatern = SearchPattern.R_EXACT_MATCH;
+				} else {
+					classNameString = null;
+					packageNameString = matchString;
+				}
+			}
+			char[] packageName = packageNameString!=null?packageNameString.toCharArray():null;
+			char[] className = classNameString!=null?classNameString.toCharArray():null;
+
+			IProject project = bp.getProject();
+			IJavaProject jp = JavaCore.create(project);
+			IJavaSearchScope scope = SearchEngine.createJavaSearchScope(new IJavaElement[] {jp});
+			SearchEngine engine = new SearchEngine();
+			engine.searchAllTypeNames(packageName, packagePatern, className, SearchPattern.R_PREFIX_MATCH, IJavaSearchConstants.CLASS, scope, new TypeNameRequestor() {
+				@Override
+				public void acceptType(
+						int modifiers,
+						char[] packageName,
+						char[] simpleTypeName,
+						char[][] enclosingTypeNames,
+						String path) {
+
+					StringBuffer fullName = new StringBuffer();
```

Comment:
You are right. And I do use StringBuilders usually I swear :) But old habits die hard and it looks like I my fingers typed buffer automaticly ;)